ksp_plugin/interface_body.hpp

```diff
@@ -595,19 +589,33 @@ inline not_null<OrbitAnalysis*> NewOrbitAnalysis(
         .subcycle = recurrence.subcycle(),
     };
   }
-  if (vessel_analysis->ground_track().has_value() &&
-      vessel_analysis->equatorial_crossings().has_value()) {
-    auto const& equatorial_crossings = *vessel_analysis->equatorial_crossings();
-    analysis->ground_track = new OrbitGroundTrack{
-        .equatorial_crossings = {
-                .longitudes_reduced_to_ascending_pass =
-                    ToInterval(equatorial_crossings.longitudes_reduced_to_pass(
-                        2 * ground_track_revolution - 1)),
-                .longitudes_reduced_to_descending_pass =
-                    ToInterval(equatorial_crossings.longitudes_reduced_to_pass(
-                        2 * ground_track_revolution)),
-            },
-    };
+  if (vessel_analysis->ground_track().has_value()) {
```

Comment:
Let's give a names to `vessel_analysis->ground_track()`, as it stands it's hard to read all these long expressions.